kafka/src/main/java/org/apache/apex/malhar/kafka/KafkaConsumerWrapper.java

```diff
@@ -137,6 +189,9 @@ public void run()
 
 
         while (wrapper.isAlive) {
+          if (wrapper.waitForReplay) {
+            Thread.sleep(100);
+          }
```

Comment:
After thread sleep, I think need to continue the loop. Otherwise, consume messages from the consumer. Code snippet for if condition be as follows:
if (wrapper.waitForReplay) {
  Thread.sleep(100);
  continue;
}